### PR TITLE
[CSL-1803] On Linux, use UTF-8 for filepath encoding

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

File: src/Database/RocksDB/Base.hs [hunk 1]
```diff
@@ -127,10 +135,34 @@ createSnapshotBracket db = allocate (createSnapshot db) (releaseSnapshot db)
...
+        initialize =
+            (, ()) <$> mkOpts opts
+        finalize (opts', ()) =
+            freeOpts opts'
+# else
+        initialize = do
+            opts' <- mkOpts opts
+            -- With LC_ALL=C, two things happen:
+            --   * rocksdb can't open a database with unicode in path;
+            --   * rocksdb can't create a folder properly.
```
Comment (gromakovsky) [inline src/Database/RocksDB/Base.hs]: 1. Is the first problem solved in this PR?
2. If it is, isn't that solution enough for the second problem?

**Comment by rockbmb** [comment]
@gromakovsky I'm not sure why Travis CI fails, I've reverted the resolver changes so there should be no reason for it to mention a missing `ghc801`. I'll continue trying to solve this problem, but the Unicode issue is now fixed on Windows.

It seems the only thing that needed to be done is have `withFilePath` introduced by @neongreen work when `ming32_HOST_OS` too, it is not necessary to even use `Win32`.

**Comment by deepfire** [comment]
@rockbmb, @gromakovsky, https://github.com/serokell/rocksdb-haskell/pull/7 is the tentative solution.

**Comment by rockbmb** [comment]
@gromakovsky all three of your concerns have now been addressed.

**Pull request merged**